ui/app/routes/allocations/allocation/task/fs.js

```diff
@@ -0,0 +1,33 @@
+import Route from '@ember/routing/route';
+import RSVP from 'rsvp';
+
+export default Route.extend({
+  model({ path = '/' }) {
+    const decodedPath = decodeURIComponent(path);
+    const task = this.modelFor('allocations.allocation.task');
+
+    const pathWithTaskName = `${task.name}${decodedPath.startsWith('/') ? '' : '/'}${decodedPath}`;
+
+    return task.stat(pathWithTaskName).then(statJson => {
```

Comment:
On all the data fetching routes, there is a custom `catch` handler to overcome Ember's poor default error handling behavior.

Example: https://github.com/hashicorp/nomad/blob/master/ui/app/routes/allocations/allocation.js#L46

The tl;dr is it sets the error state on the application controller so we can show error states while preserving the URL.

Reply:
hmm I missed this, I see… I put the error-handling in the adapter, but I’ll change it to align with the linked example, thanks.